arviz/plots/backends/bokeh/kdeplot.py

```diff
@@ -226,11 +230,16 @@ def plot_kde(
                 palette=colors,
                 **pcolormesh_kwargs
             )
+            glyphs.append(image)
             ax.x_range.range_padding = ax.y_range.range_padding = 0
 
     if backend_show(show):
         bkp.show(ax, toolbar_location="above")
-    return ax
+
+    if legend and label is not None:
+        return_glyphs = glyphs
```

Comment:
Given that glyphs is always generated, it could always be returned instead of returning none depending on the arguments. If it depends on an argument I'd make it explicit